File: golang/1.easy.two-sum/1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	diffMap := make(map[int]int)
	for i, num := range nums {
		if idx, ok := diffMap[num]; ok {
			return []int{idx, i}
		} else {
			diffMap[target-num] = i
		}
	}
	return nil
}

// @lc code=end

File: java/1.TwoSum.java
// https://leetcode-cn.com/problems/two-sum/
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], i);
        }
        for (int i = 0; i < nums.length; i++) {
            int right = target - nums[i];
            Integer pos = (Integer) map.get(right);
            if (pos != null && pos != i) {
                return new int[]{i, pos};
            }
        }
        return null;
    }
}
